### DISCUSSION TITLE
updating code/script - how/what's the process?

### DISCUSSION THREAD

**Discussion by workjoeea** [derekantrican/GAS-ICS-Sync#333] [General]
Is the code on the Github more up to date that on the Google App script link in the read me?  If so, what's the process?  I ask because it looks like some improvements have been made since I started using this (February), including stopping the removal and/or updating of previous/past events/calendar entries.  (I only just noticed today that events have been deleted, and my Google calendar only goes back maybe 4 weeks or so?...the office calendar remains untouched.)

once said update is done, how do I re-populate the previous/past events back into the calendar?

TYIA

**Comment by jonas0b1011001** [comment]
> Is the code on the Github more up to date that on the Google App script link in the read me?

Github will always be the latest codebase. The linked Project resembles the latest 'official' release that can be found [here](https://github.com/derekantrican/GAS-ICS-Sync/releases).

> If so, what's the process?

There is no convenient way to update.
1. There's a [Chrome extension](https://chrome.google.com/webstore/detail/google-apps-script-tools/iigobcdpmngdgiacebgenbccmdacnjhp) to pull the latest code, this will replace your settings in code.gs so save them beforehand.
2. Cop over the code manually.



> I only just noticed today that events have been deleted, and my Google calendar only goes back maybe 4 weeks or so?

The script will only delete events if they are no longer part of the source. Seems like your source calendar is only exporting that far back.



> how do I re-populate the previous/past events back into the calendar?

As stated above, the events don't seem to be part of the source anymore. Maybe you can manually export events earlier than what is synced. Other than that, have a look at [gcal's trash bin](https://calendar.google.com/calendar/u/0/r/trash?tab=kc), events deleted by the script will also be put there.

**Reply by workjoeea** [reply]
> > Is the code on the Github more up to date that on the Google App script link in the read me?
> 
> Github will always be the latest codebase. The linked Project resembles the latest 'official' release that can be found [here](https://github.com/derekantrican/GAS-ICS-Sync/releases).

I was able to find this, thank you though.
> 
> > If so, what's the process?
> 
> There is no convenient way to update.
> 
>     1. There's a [Chrome extension](https://chrome.google.com/webstore/detail/google-apps-script-tools/iigobcdpmngdgiacebgenbccmdacnjhp) to pull the latest code, this will replace your settings in code.gs so save them beforehand.
> 
>     2. Cop over the code manually.

Copying manually is fine by me.  Was hoping for a better way but it is what it is.  I'm guessing both the Code.js and Helpers.js, correct?
> 
> 
> > I only just noticed today that events have been deleted, and my Google calendar only goes back maybe 4 weeks or so?
> 
> The script will only delete events if they are no longer part of the source. Seems like your source calendar is only exporting that far back.

Interesting.  I'm fairly certain this is a new phenomenon.  When I started this back in February, I think everything came over from the Outlook to the Google.  Any thoughts or insight as to how or why that changed, and how to fix that?  (FWIW, I opened the link as an html in browser tab, and it does in fact stop only some weeks back, too.)

If I redo the code from the github, will that stop the deleting of old events?  And/or do I need to follow your input/advise from the thread about issue #109 ?  Possibly other issues as well?
> 
> > how do I re-populate the previous/past events back into the calendar?
> 
> As stated above, the events don't seem to be part of the source anymore. Maybe you can manually export events earlier than what is synced. Other than that, have a look at [gcal's trash bin](https://calendar.google.com/calendar/u/0/r/trash?tab=kc), events deleted by the script will also be put there.

So I tried this, to then only have the script re-delete them.  Haha.

Thank you so much for you help so far.

**Reply by jonas0b1011001** [reply]
> I'm guessing both the Code.js and Helpers.js, correct?

Usually those are the ones that get updated. However github shows when a file was last updated, better use that to reconfirm.

I have no experience with outlook, can't help you on that.
I do my best to incorporate fixes from the issues section to the master-branch, you should be good to go unless you made some niche feature addons. #109 made it to master, just set [var removePastEventsFromCalendar = false;](https://github.com/derekantrican/GAS-ICS-Sync/blob/e5cd8e02417ee38a04eadbd795201822dbac2e2d/Code.gs#L39)

> So I tried this, to then only have the script re-delete them. Haha.

True, the latest code with the settings as mentioned above will keep them after restoring from trash. Just in case you really need some of the events that recently got deleted.
